client/src/context/useAuthContext.tsx

```diff
@@ -8,7 +8,7 @@ import { User } from '../interface/User';
 import { Profile } from '../interface/Profile';
 
 interface IAuthContext {
-  profile: Profile | null | undefined;
+  profile: Profile | null | undefined | never | any;
```

Comment:
I prefer to not pollute this with `never` or `any`, look at your data coming back and make proper interfaces to capture different states of the profile so it excepts either of them